### ISSUE TITLE
Dumping PART files in C:\

### ISSUE THREAD

**Issue by SpoogLord** [datawhores/OF-Scraper#202]
**Describe the bug**
Upon failing a download the program (sometimes?) dumps a PART file in C:\

**To Reproduce**
1. Run ofscraper 
2. Fail a download (unsure how to cause a download to fail)
3. Check C:\

**Expected behavior**
Instead of dumping an unusable file in C:\ the program should just delete the failed download

**Screenshots/Logs**
![image](https://github.com/datawhores/OF-Scraper/assets/99507754/ec0f5e77-33dc-4034-8715-af1f08aec317)
https://hastebin.com/share/cuwatiyija.yaml

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro 22H2
 - Browser: Chrome
 - Version 3.0.12
- Python 
- python version: 3.11

**Additional context**
I just deleted like 120 gigs of part files that had gone unnoticed until now

**Comment by datawhores** [comment]
That was the original behavior 
But some users wanted to be able to resume files

You can turn partfileclean
To true, and the program will auto clean the files when it finishes or is forced closed

**Comment by datawhores** [comment]
Can you show your config file 
Is C your save location?

**Comment by SpoogLord** [comment]
This is my config (though I just changed partfileclean to true as per your suggestion)
![image](https://github.com/datawhores/OF-Scraper/assets/99507754/ee548912-611b-4f1b-aa2a-b4ec26b75765)

While the save location is on my C drive, it shouldn't be outputting anything to the root of the drive, right?

currently re-running with partfileclean set to true, will get back to you if it works

**Comment by SpoogLord** [comment]
Alright, the issue is now worse. For some reason ALL downloads are now failing, and for each that fails a new PART file is added to C:\. 

The previous issue started today, and I tried uninstalling and reinstalling ofscraper to see if it helped but now everything is crumbling around me?

**Comment by datawhores** [comment]
Yeah this is strange
I'll have to look at this.

But it's probably related to the placeholder class

**Comment by SpoogLord** [comment]
Update: Changing "dir_format" to C:\\Users\\Sage\\Data\\ofscraper starts outputting the part files into C:\\Users\\Sage\\Data\\ofscraper, but it still fails every download. Following that, Changing the save location to " " seems to be the same as having the dir be " " and the save location C:\\Users\\Sage\\Data\\ofscraper (?)

I believe this part of the log is the failing download process, isolated for convenience
https://hastebin.com/share/adoviximiq.lua

it seems like its not trying to make a folder to put the part file into 
it puts the part file in the root of the directory, because it cant find the folder its looking for
it looks for the image to rename it from .part to .jpeg (but cant find it because its expecting the part file to be in a folder that doesn't exist)
then the "download" fails because it cant find the file to rename it. 

Manually renaming the .part file to .jpeg DOES give a fully functional image, so the actual download itself is not failing, just the renaming of the file. Making a folder of the username of the model its trying to scrape fixes the issue on an individual model basis.

It made the folders automatically before, so I'm not sure what changed. I moved all the folders out of the save location to offload them to a NAS, but I guess in the future I just have to move the files in the folders? or copy instead of cut.

Is there some sort of cache file I can edit/delete to make it recognize the folders don't exist and tell it to remake them?

**Comment by Puk0** [comment]
Corre el programa como administrador 
Ponle ...
"threads": 1,
"key-mode-default": "keydb", y con "keydb_api": "tuApiDeKeydb",

Actualiza python a 3.11.5

**Comment by SpoogLord** [comment]
Another Update: I manually made folders for every model I was trying to scrape from, and the scraper mostly works now, but every time it tries to move on to a model I haven't scraped since making the model folders it says 
"[download.setDirectoriesDate:872]   PID: 13240 Setting Date for modified directories        download.py:872"
, freezes, and slowly starts consuming more and more RAM until it maxes my usage and my computer bluescreens. 

here's the log, but it just looks like it was working perfectly then closed normally, so its not super useful
https://hastebin.com/share/qomacepaxi.yaml

I guess the solution to this is just to keep re-running the scraper until I get through everyone I'm scraping, the presumably it will work fine in all future attempts, but this issue confuses me more than the last one did, so I'm just trying to make this issue known

And in response to the previous reply
I already run it in administrator
changing the thread count to 1 just slowed it down
changing the key mode + keydb_api broke it further, it couldn't even make a part file
and I updated python before reinstalling the scraper (which I did before making this issue, but seem to have forgot to mention)

**Comment by SpoogLord** [comment]
Update to the update:
> I guess the solution to this is just to keep re-running the scraper until I get through everyone I'm scraping, then presumably it will work fine in all future attempts
This did not work. I've been running this back to back for the last 2 days and it still gets stuck in the same way. I've uninstalled and reinstalled several times, I tried my best to manually wipe everything relating to ofscraper and reinstalling it, I've tried basically everything I can think of and the scraper just will not run to completion. 

to re-summarize my current problem: 
I run the scraper
It gets though several models
It freezes and says "[download.setDirectoriesDate:872] PID: 13240 Setting Date for modified directories download.py:872"
Then consumes more and more ram until I bluescreen or close the terminal

This is basically a completely different problem at this point, if needed I can make it a new Issue in github, just let me know

**Comment by SpoogLord** [comment]
Small addendum: checked the logs again and they were 150k lines long :)

this seems to be one download worth of errors
https://hastebin.com/share/valomudido.php

it also seems based on other errors that the scraper just isn't working atm. Until an update comes out I will stop updating this thread

**Comment by Puk0** [comment]
Cada vez que uses el -l debug borra el archivo .log sino se va hacer más grande.

![2023-09-14 22-00-38](https://github.com/datawhores/OF-Scraper/assets/138377242/6daf3a0a-176d-47bf-b155-a169bf871b11)
¿qué le pusiste otra vez cdrm2 (172.106.17.134:8080) en "key-mode-default": ?

**Comment by SpoogLord** [comment]
You told me to change that before, and I tried it, but when I did the scraper stopped making .PART files, and all downloads failed, which is worse than the issue I was having. After trying it again just now, it doesn't seem to fix the issue I'm having, It still gets stuck in the same way it was before, though there _are_ less errors in the logs

**Comment by SpoogLord** [comment]
> Until an update comes out I will stop updating this thread

Well, an update came out, and it seems to have fixed nothing (in my case at least)
I'm not 100% sure this is the same issue everyone else is having
I did fully uninstall and reinstall before trying again, though I kept my config (which looks like this [](https://hastebin.com/share/usetudepup.swift))

I decided to record a video of my current situation for clarity
[](https://youtu.be/duAbgNdT7ko)

here's the log from the recorded attempt
[](https://hastebin.com/share/ihepebeweh.yaml)

To add a little more context this was from a second attempt, which is why it looks like nothing downloaded, I had already downloaded everything from the first model in queue. If I select one model it works fine, and ends on a "Script Complete" message, If I select multiple it will often get stuck when switching to the next model, in a subsequent attempt I made it 5 models in before the freeze.

Everything I've tried _should be_ documented in comments above, this was just a post-update retry.
I know this is basically a different issue than the original issue I had, if needed I can close this post and open a new one to keep things organized.
Once again, silent until replied to or another update releases.

**Comment by SpoogLord** [comment]
this is what happens after uninstalling/reinstalling and using --output debug, the log is basically identical to before, still using --log debug

(I didn't show the full runtime this time bc it made it like 4 models in, but almost everything is the same as before)
https://github.com/datawhores/OF-Scraper/assets/99507754/3538f257-a6bf-4eaa-8ecb-6cb7282a6ce5

**Comment by Puk0** [comment]
> I did fully uninstall and reinstall before trying again, though I kept my config (which looks like this [Config](https://hastebin.com/share/usetudepup.swift))
 
- Si le pones **"manual"** te faltan private-key y client-id 
- 
        "private-key": null,
        "client-id": null,
        "key-mode-default": "manual",
        "keydb_api": " ",

- Si le pones **"keydb"** te falta keydb_api
- 
        "private-key": null,
        "client-id": null,
        "key-mode-default": "keydb",
        "keydb_api": "xxxxx",  

 keydb_api -> https://keysdb.net/api -> X-API-Key:

**Comment by SpoogLord** [comment]
@Puk0 you've tried telling me to change things involving key settings 3 times now and it has fixed nothing every time I've tried it. I'm not ignoring you but I feel like you don't know what the issue is. I can download content just fine, but the script often freezes when moving to the next model. As far as I can tell debug logging claims the script gets stuck on a local process, not a download, which logically has nothing to do with keydb.

I have made some very slight changes to my config while experimenting, just to provide as much info as I can, here is my config
[Updated Config](https://hastebin.com/share/cuzuteyila.swift)

**Comment by Puk0** [comment]
Si la siguiente cuenta tiene videos con drm si le sigues poniendo "**manual**" ocupa por fuerza la ruta de tus archivos 
**device_private_key** y **device_client_id_blob**
en "private-key" y "client-id" por eso se queda asi sin hacer nada

Por ejemplo
"private-key": "D:\\Programas\\OFR\\CDM\\**device_private_key**",
"client-id": "D:\\Programas\\OFR\\CDM\\**device_client_id_blob**",
"key-mode-default": "**manual**",

**Comment by SpoogLord** [comment]
That's why it gets stuck on "modifying date" after fully downloading everything from a model? and only once in every 4 times? I have a hard time believing that.

**Comment by Puk0** [comment]
si una cuenta **no** tiene videos con protección DRM te descarga todo (fotos/videos)

si la cuenta que sigue **si** tiene videos con DRM solo te descargara las fotos
porque ni tienes los 2 archivos device_private_key y device_client_id_blob
y tampoco si seleccionas "keydb" tienes la "keydb_api"

por eso se quedan los archivos (audio/videos) encriptados/part 
porque el programa espera la "key" para desencriptarlos y para luego unirlos

**Comment by SpoogLord** [comment]
ok, but the issue is not the part files anymore, the issue is the script freezes and bluescreens my computer. Some models I follow do have DRM protected content, and those do spit out part files, but I'm more focused on actually getting the script to fully run through all my subscriptions without crashing my pc

**Comment by SpoogLord** [comment]
I even included a video showing my issue as it was happening, RAM graph and all, did you watch it?

**Comment by Puk0** [comment]
Si lo vi
¿usas la version zip/exe, o lo instalaste con "pip"?
sino prueba con la versión "zip"

**Comment by SpoogLord** [comment]
I used pipx to install it, I could not figure out how to install the .zip version and could not find documentation on it

**Comment by Puk0** [comment]
solo baja el zip (en Assets), lo descomprimes y le das click al ofscraper_windows_… .exe

**Comment by SpoogLord** [comment]
the zip file in assets does not have a .exe file in it

**Comment by Puk0** [comment]
Porque no está disponible
solo las que dice ofscraper_windows_… zip

![2023-10-11 21-45-37](https://github.com/datawhores/OF-Scraper/assets/138377242/0524e48d-3830-4d4e-98a0-64e572480a0b)

**Comment by SpoogLord** [comment]
The issue still occurs when using the zip version

**Comment by SpoogLord** [comment]
Tiny update: it just occurred for the first time when scraping a single model, the issue seems to not be related to transitioning between models like I had previously believed, the model I scraped also had no DRM protected content, so its not that either. I'm not sure what the issue is but I believe it to be something with how the scraper installed on my pc. 

I use a custom windows installation and I'm wondering if that's the cause of these issues, maybe there's some weird stock windows dependency my installation just doesn't have (because it seems this is only happening to me). I plan to reinstall windows within the next month, I'll try again then or when there's another update

**Comment by SpoogLord** [comment]
I'm actually going to close this and recreate my current problem as a new issue, the .part files thing has been mentioned in other posts and this thread is about a completely different issue at this point.
